crates/asm/src/msg.rs

```diff
@@ -0,0 +1,23 @@
+use bitcoin_bosd::Descriptor;
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{Deserialize, Serialize};
+use strata_primitives::l1::BitcoinAmount;
+
+/// Describes an intent to withdraw funds outside of Strata
+#[derive(Clone, Debug, Eq, PartialEq, BorshSerialize, BorshDeserialize, Serialize, Deserialize)]
+pub struct WithdrawalIntent {
+    /// Quantity of L1 asset, for Bitcoin this is sats.
+    amt: BitcoinAmount,
+
+    /// Destination [`Descriptor`] for the withdrawal
+    destination: Descriptor,
+}
+
+/// Describes all the messages that can originate when processing ProtocolOps by a subprotocol that
+/// needs to be consumed by other subprotocols
+#[derive(Clone, Debug, Eq, PartialEq, BorshSerialize, BorshDeserialize, Serialize, Deserialize)]
+pub enum InterProtoMsg {
+    /// This message originates from the Core OL Subprotocol and is meant to be passed to the
+    /// Bridge Subprotocol
+    Withdrawal(WithdrawalIntent),
+}
```

Comment:
This is introducing coupling by making the general ASM infrastructure implicitly dependent on the subprotocols that we intend to use it with.  We can really leverage the type system here to avoid this.  See the discussion about `dyn` for an imperfect but better solution.

Define some kind of `SubprotoMsgRouter` type that abstracts over this low level bookkeeping, and then pass in a `&mut` into the process function instead of making it return the messages.  It allows for more abstraction over the types.